### ISSUE TITLE
Hand not letting go.

### ISSUE THREAD

**Issue by Theodorehip** [TomorrowTodayLabs/NewtonVR#108]
Often times i find that the controller wont let go of the object it's holding in the code, And you'll be able to grab that object again from anywhere. This glitch will also prevent the controller from grabbing anything else.

Anyone else having this problem?

**Comment by SmartCarrion** [comment]
Yes, and I have had that and solved it in a few ways.  There are a few cases I found that can cause this, but the root cause is that the object is still in the 'hovering' list.   'OnTriggerEnter' and 'OnTriggerExit' are used to add/remove objects to the hovering list, then the hovering list is used to see what grabbable object is closest.  

The main problem comes up because if a trigger is deactivated, it does not send 'OnTriggerExit' events.  So the key times this bug can happen are if you disable the interactable gameobject, turn off the hand trigger, or change the interactability of an interactable object.

Most of my solutions revolved around adding a 'ForceEndHovering' function and calling it from all the trouble spots.

Hope that helps, here is my function to add to NVRHand.  It may have changed the data type on something, sorry if it doesn't work out of the box, but it gives you the idea:

        public virtual void ForceEndHovering(NVRInteractable interactable)
        {
            if (interactable != null && CurrentlyHoveringOver.ContainsKey(interactable) == true)
            {
                CurrentlyHoveringOver[interactable].Clear();
                CurrentlyHoveringOver.Remove(interactable);
            }
        }

also related to issue: https://github.com/TomorrowTodayLabs/NewtonVR/issues/91

**Comment by naam00** [comment]
Thanks! Solved my problem as well.
